### PR TITLE
Added gRPC interface to collect every sketch file

### LINKED ISSUE #837: As a gRPC consumer, I want the CLI to collects all the files composing a sketch
It is already implemented by the CLI but it's not yet exposed as an API/command:

https://github.com/arduino/arduino-cli/blob/fff99af81a922eb89c01703373659b5983375093/arduino/builder/sketch.go#L100-L103

I would like to avoid duplicating code in the Pro IDE and fully rely on this CLI functionality.

Thank you!

### PR DESCRIPTION
**Pull request by silvanocerza**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?**
Exposes a gRPC interface to collect all files composing a sketch.

- **What is the current behavior?**
gRPC interface is not exposed.

* **What is the new behavior?**
gRPC consumers have a way to collect all files composing a sketch.

- **Does this PR introduce a breaking change?**
Nope.

* **Other information**:
Fixes #837.

---

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

### CODE REVIEW

**Comment by silvanocerza** [comment]
Codecov link is failing again, the rest of the tests are fine. :shrug:

File: rpc/commands/commands.proto [hunk 1]
```diff
@@ -218,10 +221,10 @@ message UpgradeReq {
...
 
 message UpgradeResp {
-	// Progress of the downloads of the platforms and libraries files.
-	DownloadProgress progress = 1;
-	// Description of the current stage of the upgrade.
-	TaskProgress task_progress = 2;
+  // Progress of the downloads of the platforms and libraries files.
+  DownloadProgress progress = 2;
+  // Description of the current stage of the upgrade.
+  TaskProgress task_progress = 3;
```
Comment (kittaakos) [inline rpc/commands/commands.proto:227]: Why this change?

File: commands/instances.go [hunk 2]
```diff
@@ -676,3 +677,28 @@ func createInstance(ctx context.Context, getLibOnly bool) (*createInstanceResult
 
 	return res, nil
 }
+
+// SketchLoad collects and returns all files composing a sketch
+func SketchLoad(ctx context.Context, req *rpc.SketchLoadReq) (*rpc.SketchLoadResp, error) {
+	sketch, err := builder.SketchLoad(req.SketchPath, "")
+	if err != nil {
+		return nil, fmt.Errorf("Error loading sketch: %v", err)
```
Comment (kittaakos) [inline commands/instances.go:685]: Can we make `req.SketchPath` as part of the error message?

File: rpc/commands/commands.proto [hunk 3]
```diff
@@ -58,6 +58,9 @@ service ArduinoCore {
   // Get the version of Arduino CLI in use.
   rpc Version(VersionReq) returns (VersionResp) {}
 
+  // Returns all files composing a Sketch
+  rpc SketchLoad(SketchLoadReq) returns (SketchLoadResp) {}
```
Comment (kittaakos) [inline rpc/commands/commands.proto:62]: [minor]: Why not `LoadSketch`? (verb + object)

**Review by kittaakos — COMMENTED** [review]
I am going to try how it behaves in the IDE. Thank you for creating the PR 👍

File: commands/instances.go (hunk 2 above)
Comment (silvanocerza) [reply commands/instances.go:685]: Yes, we could but think that some informationg is already in `err`.

File: rpc/commands/commands.proto (hunk 1 above)
Comment (silvanocerza) [reply rpc/commands/commands.proto:227]: No idea, I didn't manually change it, it's probably some auto formatting that I missed.

File: rpc/commands/commands.proto (hunk 3 above)
Comment (silvanocerza) [reply rpc/commands/commands.proto:62]: I've used the same name that's used in the `builder` package: https://github.com/arduino/arduino-cli/blob/master/arduino/builder/sketch.go#L103

I can change it `LoadSketch` with no issues by the way, I like the suggestion.

File: rpc/commands/commands.proto (hunk 1 above)
Comment (kittaakos) [reply rpc/commands/commands.proto:227]: It's not about the indentation; this is a change from `1` to `2`, and from `2` to `3`. Just asking though...

Comment (silvanocerza) [reply rpc/commands/commands.proto:227]: Yes, I noticed it. Still not sure how it did happen. :shrug:

File: rpc/commands/commands.proto [hunk 4]
```diff
@@ -230,3 +233,19 @@ message VersionResp {
   // The version of Arduino CLI in use.
   string version = 1;
 }
+
+message SketchLoadReq {
+  // Absolute path to single sketch file or a sketch folder
+  string sketch_path = 1;
+}
```
Comment (kittaakos) [inline rpc/commands/commands.proto:240]: This should also include the `instance`.
```
message SketchLoadReq {
  Instance instance = 1;
  // Absolute path to single sketch file or a sketch folder
  string sketch_path = 2;
}
```

Comment (silvanocerza) [reply rpc/commands/commands.proto:240]: Sure!

Comment (kittaakos) [reply rpc/commands/commands.proto:240]: Well, I do not know if it's required, but strange to not having symmetric APIs. @cmaglie, can you please help with this? Thank you!

Comment (silvanocerza) [reply rpc/commands/commands.proto:240]: I didn't include it because it's actually not used by that command but I guess it costs nothing adding it.

Comment (kittaakos) [reply rpc/commands/commands.proto:240]: Works well so far, great simplification for the IDE. Thanks!

**Comment by silvanocerza** [comment]
Fixed those issues you pointed out @kittaakos.

**Review by rsora — APPROVED** [review]
